ktor-http/ktor-http-cio/common/src/io/ktor/http/cio/HttpParser.kt

```diff
@@ -116,14 +116,16 @@ internal suspend fun parseHeaders(
             val nameHash = builder.hashCodeLowerCase(nameStart, nameEnd)
 
             val headerEnd = range.end
-            parseHeaderValue(builder, range)
+            val shouldSkipHeader = parseHeaderValue(builder, range)
 
             val valueStart = range.start
             val valueEnd = range.end
             val valueHash = builder.hashCodeLowerCase(valueStart, valueEnd)
             range.start = headerEnd
 
-            headers.put(nameHash, valueHash, nameStart, nameEnd, valueStart, valueEnd)
+            if (!shouldSkipHeader) {
+                headers.put(nameHash, valueHash, nameStart, nameEnd, valueStart, valueEnd)
+            }
```

Comment:
We need some kind diagnostics here, like logging on trace level. Not sure what logger we could use here....